src/Export/ExportItemAdapter.php

```diff
@@ -55,12 +55,49 @@ public function adapt(Item $item, ProductEntity $product): Item
     {
         $this->eventDispatcher->dispatch(new BeforeItemAdaptEvent($product, $item), BeforeItemAdaptEvent::NAME);
 
-        $item->setName($this->adapterFactory->getNameAdapter()->adapt($product));
+        if ($name = $this->adapterFactory->getNameAdapter()->adapt($product)) {
+            $item->setName($name);
+        }
+
         foreach ($this->adapterFactory->getAttributeAdapter()->adapt($product) as $attribute) {
             $item->addMergedAttribute($attribute);
         }
+
         $item->setAllPrices($this->adapterFactory->getPriceAdapter()->adapt($product));
-        $item->setUrl($this->adapterFactory->getUrlAdapter()->adapt($product));
+
+        if ($url = $this->adapterFactory->getUrlAdapter()->adapt($product)) {
+            $item->setUrl($url);
+        }
+
+        if ($description = $this->adapterFactory->getDescriptionAdapter()->adapt($product)) {
+            $item->setDescription($description);
+        }
+
+        if ($dateAdded = $this->adapterFactory->getDateAddedAdapter()->adapt($product)) {
+            $item->setDateAdded($dateAdded);
+        }
+
+        foreach ($this->adapterFactory->getKeywordsAdapter()->adapt($product) as $keyword) {
+            $item->addKeyword($keyword);
+        }
+
+        foreach ($this->adapterFactory->getImagesAdapter()->adapt($product) as $image) {
+            $item->addImage($image);
+        }
+
+        $item->setSalesFrequency($this->adapterFactory->getSalesFrequencyAdapter()->adapt($product));
```

Comment:
We should also move this inside a nullable check, as the return type should be nullable